cabal-install/Distribution/Client/CmdClean.hs

```diff
@@ -0,0 +1,125 @@
+-- | cabal-install CLI command: clean
+--
+module Distribution.Client.CmdClean (
+    cleanCommand,
+    cleanAction,
+  ) where
+
+import Distribution.Client.DistDirLayout
+         ( DistDirLayout(..), defaultDistDirLayout )
+import Distribution.Client.ProjectConfig
+         ( findProjectRoot )
+import qualified Distribution.Client.Setup as Client
+import Distribution.Client.Setup
+         ( GlobalFlags )
+import Distribution.PackageDescription
+         ( emptyHookedBuildInfo, updatePackageDescription )
+import Distribution.PackageDescription.Configuration
+         ( flattenPackageDescription )
+import Distribution.PackageDescription.Parse
+         ( readPackageDescription )
+import Distribution.Simple.Command
+         ( CommandUI(..), noExtraFlags, option, usageAlternatives )
+import Distribution.Simple.Configure
+         ( findDistPrefOrDefault, maybeGetPersistBuildConfig
+         , writePersistBuildConfig )
+import Distribution.Simple.Setup
+         ( CleanFlags(..), defaultCleanFlags, defaultDistPref, fromFlag
+         , fromFlagOrDefault, optionDistPref, optionVerbosity, toFlag, trueArg )
+import Distribution.Simple.Utils
+         ( chattyTry, defaultPackageDesc, notice )
+import Distribution.Types.PackageDescription
+         ( PackageDescription(..) )
+
+import Control.Monad (when)
+import Data.Foldable (forM_, traverse_)
+import System.Directory
+import System.FilePath ((</>))
+
+cleanCommand :: CommandUI CleanFlags
+cleanCommand = Client.installCommand {
+  commandName         = "new-clean",
+  commandSynopsis     = "Cleans up after a Nix-local build.",
+  commandUsage        = usageAlternatives "new-clean" [ "[FLAGS]" ],
+  commandDescription  = Just $ \_ ->
+    "Cleans up Nix-local build files such as .hi, .o, preprocessed sources, etc.",
```

Comment:
Since you already mentioned nix-local-build in the synopsis, I think you can just say "intermediate build files".